src/rust/Cargo.toml

```diff
@@ -0,0 +1,22 @@
+[package]
+name = "mixxx-rustlib"
```

Comment:
My personal opinion is that it would be better to wrap rekordcrate in a C or C++ interface outside of Mixxx and then link that wrapped library in instead of adding Rust code to Mixxx itself. This would make it easy to gate the Rekordcrate support behind a feature flag too, which IMO is a good idea anyway (we've done so for a number of other libraries, e.g. Rubberband in #12661).

While Rust is a cool language and I'd love to see parts of Mixxx written in it, integrating it into Mixxx's core toolchain opens a huge can of worms:

- Adding a new language with a wildly different ecosystem will necessarily add complexity, both to the codebase and the build system
- Interaction between Rust/C++/Qt involves requires pulling in more layers of dependencies
- Building Mixxx, especially in a cross-compilation scenario (such as our GitHub Actions build targeting ARM macOS or a future iOS version) would now require not just a C++ toolchain, but also a Rust toolchain with the correct target toolchain. This will likely complicate the build setup quite significantly for anything outside of the standard host to host compilation scenario... I'm not sure how the interaction between CMake, vcpkg and Cargo handles this by default.

Keeping the Rust code encapsulated in an out-of-tree C/C++-wrapped library would contain this complexity in a way that's much easier to manage (e.g. cross-building a Rust library packaged up in a vcpkg port should be pretty easy, but hacking together the Mixxx build system to cross-build a mixed Rust/C++ codebase for a new target will always be a non-trivial endeavor.

Reply:
Ok I get the argument. So I suppose the code should live in mixxxdj/mixxx-rekordboxlib or something similar?

Reply:
Let's first discuss this with the other devs on Zulip, before spending work on this.